pkg/signup/service/signup_service.go

```diff
@@ -467,6 +483,23 @@ func (s *ServiceImpl) DoGetSignup(ctx *gin.Context, provider ResourceProvider, u
 		Message:              murCondition.Message,
 		VerificationRequired: states.VerificationRequired(userSignup),
 	}
+
+	if mur.Status.ProvisionedTime != nil {
+		startDateValue := mur.Status.ProvisionedTime.Format(ISO8601Format)
+		signupResponse.StartDate = &startDateValue
+
+		// #### The rest of this code only makes sense when there is a ProvisionedTime set, so we include it in the same block ####
+
+		if !userSignup.Status.ScheduledDeactivationTimestamp.IsZero() {
+			endDateValue := userSignup.Status.ScheduledDeactivationTimestamp.Format(ISO8601Format)
+			signupResponse.EndDate = &endDateValue
+
+			// Calculate the number of days remaining
+			daysRemaining := time.Until(userSignup.Status.ScheduledDeactivationTimestamp.Time).Hours() / 24
+			signupResponse.DaysRemaining = &daysRemaining
```

Comment:
this is already calculated above, why do you need to recalculate it here again?

Reply:
Oops, major oversight.. I must have written the above code previously and forgotten about it.